index.d.ts

```diff
@@ -1,17 +1,20 @@
-import { Dispatch, Store, Unsubscribe } from 'redux';
+import { Dispatch, Store, Unsubscribe, AnyAction } from 'redux';
 
 export function shallowEquals(a: any, b: any): boolean;
 
-interface Options {
+type Options = {
   skipInitialCall?: boolean;
   equals?: typeof shallowEquals;
 }
 
-type Observer<S = any, MS = any> = unknown;
-type Mapper<S = any, MS = any> = (state: S) => MS;
-type Dispatcher<MS> = (dispatch: Dispatch<MS>, currentState: MS, previousState: MS) => unknown;
+type Mapper<S, MS> = (state: S) => MS;
+type Observer<S = any> = (state: S, dispatch: Dispatch<AnyAction>) => void;
+type Dispatcher<MS> = (dispatch: Dispatch<AnyAction>, current: MS, previous?: MS) => void;
```

Comment:
`Dispatch<AnyAction>` works with both Redux `3.7.2` and `4.0.4`. I think `Dispatch<MS>` is incorrect in the previous commit. Also, both the `Observer` and `Dispatcher` return nothing.

`state` is mandatory here, so shouldn't `Observer<S = any>` actually be `Observer<S>`?

Reply:
To explain, Dispatcher<MS> should be meaningful for `@types/redux`, which is a third-party maintained type and differ from the current official typing (and that's why we need some updates here)